js/msp/MSPHelper.js

```diff
@@ -0,0 +1,1811 @@
+'use strict';
+
+
+function MspHelper () {
+    MspHelper.prototype.process_data = function(dataHandler) {
```

Comment:
Sorry to reply by email, but GitHub seems to be 'eating' comments on
code when a rebase is done. :-(

On Fri, Aug 12, 2016 at 8:16 AM, Anton Stålheim
<notifications@github.com> wrote:
>
> Ok fixed this. Haven't really worked with "oop" for javascript earlier and still learning. What is the benefit of defining it likes this? Seems a bit more confusing then defining the methods within the class declaration as in other languages.

The reason for this is that javascript's way of object orientation is
different from the 'classical', class based model. I.e. there are no
classes in JS, only objects. What you think of as the 'class' above is
just the object's constructor function that is called on `new
MspHelper()`. In that sense, it is a cleaner approach to statically
add the methods to the prototype (methods on the prototype will be
referenced by any objects created by `new...`). Also, not adding them
in the constructor means they are defined only once, which is more in
line with how class methods are defined in 'classical' OO.

If you want to get a good understanding of the underlying concepts of
JS, and can spare a couple of (quite entertaining) hours, I can highly
recommend this: https://www.youtube.com/watch?v=JxAXlJEmNMg